src/core/requests/send_chat_action.rs

```diff
@@ -1,8 +1,92 @@
-use crate::core::requests::RequestContext;
-//TODO:: need implementation
+use crate::core::network;
+use crate::core::requests::{ChatId, Request, RequestContext, RequestFuture, ResponseResult};
+use crate::core::types::Message;
 
+///Use this method when you need to tell the user that something is happening on the bot's side.
+///The status is set for 5 seconds or less (when a message arrives from your bot, Telegram clients clear its typing status).
+///Returns True on success.
 #[derive(Debug, Clone, Serialize)]
 struct SendChatAction<'a> {
     #[serde(skip_serializing)]
     ctx: RequestContext<'a>,
+    /// Unique identifier for the target chat or
+    /// username of the target channel (in the format @channelusername)
+    pub chat_id: ChatId,
+    /// Type of action to broadcast. Choose one, depending on what the user is
+    /// about to receive: typing for text messages, upload_photo for photos,
+    /// record_video or upload_video for videos, record_audio or upload_audio
+    /// for audio files, upload_document for general files, find_location for
+    /// location data, record_video_note or upload_video_note for video notes.
+    pub action: ChatAction,
 }
+
+#[derive(Debug, Serialize, From, Clone)]
+enum ChatAction {
```

Comment:
Use the #[serde(rename_all="snake_case")]

Reply:
fix